### PR TITLE
Feature/compliance year tests

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
change to remember selected compliance year and related tests

### CODE REVIEW

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/ManageEvidenceNotesControllerReviewEvidenceNoteTests.cs [hunk 1]
```diff
@@ -291,41 +314,30 @@ public async Task ReviewEvidenceNotePost_GivenInvalidModel_BreadcrumbShouldBeSet
...
             //arrange
-            var httpContext = new HttpContextMocker();
-            httpContext.AttachToController(ManageEvidenceController);
+            var organisationId = Fixture.Create<Guid>();
+            var evidenceNoteId = Fixture.Create<Guid>();
+            var complianceYear = Fixture.Create<int>();
             var model = GetValidModel();
+            model.ViewEvidenceNoteViewModel.OrganisationId = organisationId;
+            model.ViewEvidenceNoteViewModel.Id = evidenceNoteId;
+            model.ViewEvidenceNoteViewModel.SelectedComplianceYear = complianceYear;
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/ManageEvidenceNotesControllerReviewEvidenceNoteTests.cs:327]: kinda related to compliance year, there is this ComplianceYearDisplay in ViewEvidenceNoteViewModel class. which gets StartDate. Does it need to be updated to get the SelectedComplianceYear ?

File: src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs [hunk 2]
```diff
@@ -48,5 +48,7 @@ public class ViewEvidenceNoteViewModel : EvidenceNoteViewModel
         public bool DisplayAatfName { get; set; }
 
         public bool DisplayH2Title { get; set; }
+
+        public int SelectedComplianceYear { get; set; }
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs:52]: can you double check line 21 in this model, does         [DisplayName("Compliance year")]
        public string ComplianceYearDisplay => StartDate.Value.Year.ToString(); need updating?

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/ManageEvidenceNotesControllerReviewEvidenceNoteTests.cs [hunk 3]
```diff
@@ -108,16 +108,19 @@ public async Task DownloadEvidenceGet_GivenEvidenceNote_ModelMapperShouldBeCalle
...
             // act
-            await ManageEvidenceController.DownloadEvidenceNote(OrganisationId, EvidenceNoteId);
+            await ManageEvidenceController.DownloadEvidenceNote(OrganisationId, EvidenceNoteId, complianceYear);
 
             // assert
             A.CallTo(() => Mapper.Map<ViewEvidenceNoteViewModel>(
-                A<ViewEvidenceNoteMapTransfer>.That.Matches(v => v.EvidenceNoteData.Equals(noteData) && v.SchemeId.Equals(OrganisationId)))).MustHaveHappenedOnceExactly();
+                A<ViewEvidenceNoteMapTransfer>.That.Matches(v => v.EvidenceNoteData.Equals(noteData) && 
+                                                                 v.SchemeId.Equals(OrganisationId) && 
+                                                                 v.SelectedComplianceYear.Equals(complianceYear)))).MustHaveHappenedOnceExactly();
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/ManageEvidenceNotesControllerReviewEvidenceNoteTests.cs:123]: very minor thing. it's an int value so == should be fine

**Comment by meglenapnovoselska** [comment]
not massively important but there is this ReviewEvidenceNotesViewModelMapTests, using both of the models you updated - ViewEvidenceNoteMapTransfer and  ViewEvidenceNoteViewModel, could add a test there for .that.Mathes the selected compliance year, up to you.

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs (hunk 2 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/ViewModels/Shared/ViewEvidenceNoteViewModel.cs:52]: Yup will do another change to bring the compliance year from the db

File: src/EA.Weee.Web/Areas/Scheme/Views/ManageEvidenceNotes/_ManageEvidenceNotesHeaderAndTabs.cshtml [hunk 4]
```diff
@@ -22,7 +22,7 @@
     </h1>
 </header>
 
- @{ Html.RenderPartial("_ComplianceYearPartial", Model.ManageEvidenceNoteViewModel, new ViewDataDictionary() {{ "controller", "ManageEvidenceNotes" }, { "tab", Model.ActiveDisplayOption.ToDisplayString() }});}
+ @{ Html.RenderPartial("_ComplianceYearPartial", Model.ManageEvidenceNoteViewModel, new ViewDataDictionary() {{ "controller", "ManageEvidenceNotes" }, { "tab", Model.ActiveDisplayOption.ToDisplayString() }, { "width", "107px" } });}
```
[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.Web/Areas/Scheme/Views/ManageEvidenceNotes/_ManageEvidenceNotesHeaderAndTabs.cshtml:25]: that's fine as long as Sarah is happy

File: src/EA.Weee.Web/Views/Shared/_ComplianceYearPartial.cshtml [hunk 5]
```diff
@@ -4,16 +4,17 @@
...
 {
     @Html.Hidden("tab", selectedTab)
     <span class="govuk-body"><strong>@Html.LabelFor(m => m.SelectedComplianceYear)</strong></span>
     <span class="text-margin">
-        @Html.Gds().DropDownListFor(m => m.SelectedComplianceYear, 
-            new SelectList(Model.ComplianceYearList), 
-            new { autocomplete = "false", style="min-width: 100px;"}, false, false)
+        @Html.Gds().DropDownListFor(m => m.SelectedComplianceYear,
+            new SelectList(Model.ComplianceYearList),
+            new { autocomplete = "false", style="min-width: " + width }, false, false)
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Views/Shared/_ComplianceYearPartial.cshtml:17]: I did not :) but unfortunately fixed the width as Sarah would like it lined up with the tab

[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.Web/Views/Shared/_ComplianceYearPartial.cshtml:17]: okay :)

**Pull request merged**